packages/botonic-plugin-flow-builder/src/action/intent.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  EventBotAiModel,
+  EventName,
+} from '@botonic/plugin-hubtype-analytics/lib/cjs/types'
+import { ActionRequest } from '@botonic/react'
+
+import { FlowBuilderApi } from '../api'
+import {
+  HtIntentNode,
+  HtNodeWithContent,
+} from '../content-fields/hubtype-fields'
+import { trackEvent } from './tracking'
+
+export async function getNodeByIntent(
+  cmsApi: FlowBuilderApi,
+  locale: string,
+  request: ActionRequest
+) {
+  const intentNode = cmsApi.getIntentNode(request.input, locale)
+  const eventBotAiModel: EventBotAiModel = {
+    event_type: EventName.botAiModel,
+    event_data: {
+      intent: request.input.intent as string,
+      confidence: request.input.confidence as number,
+      confidence_successful: true,
+    },
+  }
+  if (isIntentValid(intentNode, request, cmsApi) && intentNode?.target?.id) {
+    await trackEvent(request, eventBotAiModel)
+    return cmsApi.getNodeById<HtNodeWithContent>(intentNode.target.id)
+  } else {
+    eventBotAiModel.event_data.confidence_successful = false
```

Comment:
beware modifying directly the attribute of the object unless this is the behavior you expect for this object. ow, you should use something to clone the data in order to have a "stable" object.

Reply:
I have done so because if the intentNode is valid (confidence higher than expected) I want to track with confidence_successful: true. Otherwise this attribute has to be false. 
This way I don't have to create two identical objects with only one attribute changing.